### PR TITLE
Productionize ground server

### PR DESCRIPTION
**Pull request by ZachGarcia42**

### CODE REVIEW

**Comment by ZachGarcia42** [comment]
* **#148** <a href="https://app.graphite.dev/github/pr/cornellrocketryteam/Ground-Software/148?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @ZachGarcia42 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: ground-server/src/app/page.tsx [hunk 1]
```diff
@@ -64,27 +64,45 @@ export default function Home() {
...
+            setTimeout(() => connect(fallbackUrl), 1000); //Retry after a delay
 
-    wsRef.current.onerror = (error) => {
-      console.error("WebSocket error:", error);
+          } else {
+            console.error("WebSocket connection failed permanently.")
+          }
+        };
+      } catch (error) {
+        console.error("Error creating WebSocket:", error);
```
Comment (maxslarsson) [inline ground-server/src/app/page.tsx:93]:
```suggestion
```

File: ground-server/src/app/page.tsx [hunk 2]
```diff
@@ -64,27 +64,45 @@ export default function Home() {
...
   useEffect(() => {
-    wsRef.current = new WebSocket("ws://localhost:8080/ws"); // Replace with your WebSocket server URL
+    const primaryUrl = "ws://10.48.59.182:8080/ws";
+    const fallbackUrl = "ws://localhost:8080/ws";
 
-    wsRef.current.onopen = () => {
-      console.log("WebSocket connection opened");
-    };
+    const connect = (url: string) => {
+      try {
```
Comment (maxslarsson) [inline ground-server/src/app/page.tsx:71]:
```suggestion
```

File: ground-server/src/app/page.tsx [hunk 3]
```diff
@@ -64,27 +64,45 @@ export default function Home() {
...
+
+        wsRef.current.onclose = (event) => {
+          console.log("WebSocket connection closed:", event.reason, event.code);
+        };
+
+        wsRef.current.onerror = (error) => {
+          console.error("WebSocket error:", error);
+          if (url === primaryUrl) {
+            console.log(`Attempting fallback to ${fallbackUrl}`);
+            setTimeout(() => connect(fallbackUrl), 1000); //Retry after a delay
```
Comment (maxslarsson) [inline ground-server/src/app/page.tsx:86]:
```suggestion
            connect(fallbackUrl);
```

**Review by maxslarsson — CHANGES_REQUESTED** [review]

**Review by maxslarsson — APPROVED** [review]

**Comment by ZachGarcia42** [comment]
### Merge activity

* **Nov 16, 9:35 PM EST**: A user merged this pull request with [Graphite](https://app.graphite.dev/github/pr/cornellrocketryteam/Ground-Software/148).

**Pull request merged**